transport.go

```diff
@@ -70,17 +71,15 @@ func getRequestBodyFromEvent(event *Event) []byte {
 		return body
 	}
 
-	partialMarshallMessage := "Original event couldn't be marshalled. Succeeded by stripping the data " +
-		"that uses interface{} type. Please verify that the data you attach to the scope is serializable."
-	// Try to serialize the event, with all the contextual data that allows for interface{} stripped.
-	event.Breadcrumbs = nil
-	event.Contexts = nil
-	event.Extra = map[string]interface{}{
-		"info": partialMarshallMessage,
+	// Try to serialize the event, with all unmarshalable data stripped
+	for i, crumb := range event.Breadcrumbs {
+		safeMarshal(fmt.Sprintf("Breadcrumb[%d].Data", i), crumb.Data)
 	}
+	safeMarshal("Contexts", event.Contexts)
+	safeMarshal("Extra", event.Extra)
```

Comment:
We could refactor this to make "partial marshalling" part of `Event.MarshalJSON` and `Breadcrumb.MarshalJSON`.